### PR TITLE
chore: autogenerate audit log documentation

### LINKED ISSUE #5356: audit: autogenerate docs for all events that are audited
Let's add a table to our auto-generated docs based on the events we audit. This will make it easier for us to smoke-test the audit logs, identify actions that are not audited, and give customers accurate information on what we're auditing.

We're currently doing this in our docs for our Prometheus metrics

https://coder.com/docs/coder-oss/latest/admin/prometheus#available-metrics

### PR DESCRIPTION
**Pull request by Kira-Pilot**
Resolves #5356
We now autogenerate the resource table in our Audit Log documentation:
https://coder.com/docs/v2/0facafef8f0ea6e3ba065a79993c622fb378f108/admin/audit-logs

### CODE REVIEW

**Comment by Kira-Pilot** [comment]
@kylecarbs The below check is failing. Shall I add an exception to `scripts/check_enterprise_imports.sh` for this import or move the script elsewhere?
Screen Shot 2023-01-25 at 3 47 14 PM

File: scripts/auditdocgen/main.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+	var buffer bytes.Buffer
+	buffer.Write(doc[:tableStartIndex])
+	buffer.WriteByte('\n')
+
+	buffer.WriteString("|<b>Resource<b>||\n")
+	buffer.WriteString("|--|-----------------|\n")
+
+	for resourceName, resourceFields := range auditableResourcesMap {
+		buffer.Write([]byte("|" + resourceName + "|<table><thead><tr><th>Field</th><th>Tracked</th></tr></thead><tbody>"))
```
Comment (mtojek) [inline scripts/auditdocgen/main.go:106]: buffer.WriteString ?

File: scripts/auditdocgen/main.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+		for fieldName, isTracked := range resourceFields {
+			buffer.Write([]byte("<tr><td>" + fieldName + "</td><td>" + strconv.FormatBool(isTracked) + "</td></tr>"))
```
Comment (mtojek) [inline scripts/auditdocgen/main.go:109]: buffer.WriteString ?

**Review by mtojek — COMMENTED** [review]
I found only minor issues, so we can merge it soon 👍 

> ERROR: AGPL code cannot import enterprise!

Yup, an exception would be enough. You can include it in this PR, Kira.

File: docs/admin/audit-logs.md [hunk 3]
```diff
@@ -5,15 +5,23 @@ their deployment.
```
[the remark this answers is not in the document]
Comment (Kira-Pilot) [reply docs/admin/audit-logs.md:5]: It's in there, I think: https://github.com/coder/coder/blob/main/docs/manifest.json#L258

File: docs/admin/audit-logs.md [hunk 4]
```diff
@@ -5,15 +5,23 @@ their deployment.
...
+| GitSSHKey          | <table><thead><tr><th>Field</th><th>Tracked</th></tr></thead><tbody><tr><td>public_key</td><td>true</td></tr><tr><td>user_id</td><td>true</td></tr><tr><td>created_at</td><td>false</td></tr><tr><td>updated_at</td><td>false</td></tr><tr><td>private_key</td><td>true</td></tr></tbody></table>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
```
[the remark this answers is not in the document]
Comment (Kira-Pilot) [reply docs/admin/audit-logs.md:14]: Ben should definitely weigh in if he has more thoughts, but I showed him the site preview yesterday and you can view it as well: https://coder.com/docs/v2/0facafef8f0ea6e3ba065a79993c622fb378f108/admin/audit-logs

I think it renders pretty well! I might submit a follow-up PR to add some additional information.

File: scripts/auditdocgen/main.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+			fieldNameMap[fieldName] = action != audit.ActionIgnore
+			auditableResourcesMap[friendlyResourceName] = fieldNameMap
+		}
+	}
+
+	return auditableResourcesMap, nil
+}
+
+// Reads the content of docs/admin/audit-logs.md
+func readAuditDoc() ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (Kira-Pilot) [reply scripts/auditdocgen/main.go:75]: Speak for yourself :D 

JK, will take out

**Review by kylecarbs — APPROVED** [review]

File: scripts/check_enterprise_imports.sh [hunk 6]
```diff
@@ -9,7 +9,7 @@ source "$(dirname "${BASH_SOURCE[0]}")/lib.sh"
 cdroot
 
 set +e
-find . -regex ".*\.go" | grep -v "./enterprise" | xargs grep -n "github.com/coder/coder/enterprise"
+find . -regex ".*\.go" | grep -v "./enterprise" | grep -v "./scripts/auditdocgen/main.go" | xargs grep -n "github.com/coder/coder/enterprise"
```
Comment (mtojek) [inline scripts/check_enterprise_imports.sh:12]: `grep -v "./scripts/auditdocgen/*.go"` - in case we add more .go files in the future.

Comment (Kira-Pilot) [reply scripts/check_enterprise_imports.sh:12]: Good idea! I'll add this change in [this PR](https://github.com/coder/coder/pull/5887).

**Pull request merged**